File: clients/intellij-community/src/test/java/dev/cleanroommc/workbench/intellij/community/CurrentContextFeatureClientTest.java
package dev.cleanroommc.workbench.intellij.community;

import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

final class CurrentContextFeatureClientTest {
    @Test
    void actionsContainNoWorkspacePathOrOwnerIdentity() {
        for (String action : List.of(
                "open", "test", "apply", "verify", "rollback", "recover"
        )) {
            assertEquals(
                    List.of("change", "material-fluid-recipe", action, "--json"),
                    CurrentContextFeatureClient.arguments(action)
            );
        }
        assertThrows(
                IllegalArgumentException.class,
                () -> CurrentContextFeatureClient.arguments("start")
        );
        assertThrows(
                IllegalArgumentException.class,
                () -> CurrentContextFeatureClient.arguments(
                        "work-session-v2-" + "1".repeat(32)
                )
        );
    }
}

File: clients/vscode/test/currentContextFeatureClient.test.js
"use strict";

const assert = require("node:assert/strict");
const test = require("node:test");

const {
  currentContextFeatureArguments,
  invokeCurrentContextFeatureAction,
} = require("../currentContextFeatureClient");

test("current-context actions contain no workspace, path, or owner identity", () => {
  for (const action of ["open", "test", "apply", "verify", "rollback", "recover"]) {
    assert.deepEqual(
      [...currentContextFeatureArguments(action)],
      ["change", "material-fluid-recipe", action, "--json"],
    );
  }
  for (const rejected of ["start", "", "work-session-v2-" + "1".repeat(32), null]) {
    assert.throws(
      () => currentContextFeatureArguments(rejected),
      /unsupported/,
    );
  }
});

test("installed entrypoint returns exact child process custody", async () => {
  const script = "process.stdout.write(JSON.stringify({state:'retained',argv:process.argv.slice(1)}))";
  const result = await invokeCurrentContextFeatureAction(process.execPath, "open", {
    launch: {
      configured: process.execPath,
      coreExecutable: process.execPath,
      distribution: null,
      executable: process.execPath,
      host: "native",
      prefixArguments: ["-e", script, "--"],
    },
    timeoutMs: 10_000,
  });
  assert.equal(result.format, "workbench-current-context-feature-action-v1");
  assert.deepEqual(
    result.outcome.argv,
    ["change", "material-fluid-recipe", "open", "--json"],
  );
  assert.deepEqual(
    [...result.arguments],
    ["change", "material-fluid-recipe", "open", "--json"],
  );
  assert.ok(Number.isSafeInteger(result.invocation.pid) && result.invocation.pid > 0);
  if (process.platform === "linux") {
    assert.ok(
      Number.isSafeInteger(result.invocation.process_start_time_ticks)
        && result.invocation.process_start_time_ticks > 0,
    );
  }
  assert.equal(result.invocation.exit_code, 0);
  assert.equal(result.invocation.signal, null);
  assert.equal(result.invocation.stderr, "");
  assert.match(result.invocation.started_at, /Z$/);
  assert.match(result.invocation.finished_at, /Z$/);
});
